### PR TITLE
[Requirements] Add a lock-file for mlrun-api Python dependencies

### PR DESCRIPTION
**Pull request by jond01**
Use uv 0.5.5: https://docs.astral.sh/uv/.

Implements [ML-8383](https://iguazio.atlassian.net/browse/ML-8383).

Apart from locking the dependencies for reproducibility, it also:
* Significantly accelerates the `make api` **build time** to 30% of the time with pip (no cache).
* Reduces the **image size** to 68% the size with pip.

The `mlrun/mlrun-api` Docker image figures:

| tool                 | pip  | uv   |
|----------------------|------|------|
| build time (minutes) | 4:55 | 1:29 |
| size (GB)            | 4.37 | 2.96 |

From my local testing - the Docker image works perfectly fine.

[ML-4648]: https://iguazio.atlassian.net/browse/ML-4648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ML-8383]: https://iguazio.atlassian.net/browse/ML-8383?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -775,3 +775,15 @@ ifdef MLRUN_DOCKER_CACHE_FROM_TAG
...
+.PHONY: update-mlrun-api-deps
+update-api-deps: ## Update mlrun-api locked requirements file
+	uv pip compile \
+	requirements.txt \
+	extras-requirements.txt \
+	dockerfiles/mlrun-api/requirements.txt \
+	--python-version $(MLRUN_PYTHON_VERSION) \
+	--universal \
+	--generate-hashes \
+	-U -o dockerfiles/mlrun-api/locked-requirements.txt
```
Comment (liranbg) [inline Makefile]: add condition to update lock file when adding specific requirement

Comment (jond01) [reply Makefile]: Done in 218007b

File: Makefile [hunk 2]
```diff
@@ -782,3 +786,23 @@ ifdef MLRUN_DOCKER_CACHE_FROM_TAG
...
+	uv_version=$$(uv version | cut -d' ' -f2); \
+	result=$$(pysemver compare $$uv_version $(MLRUN_UV_VERSION)); \
+	if [ "$$result" -eq -1 ]; then \
+	  echo "Error: The running uv version ($$uv_version) is outdated. Upgrade uv to version $(MLRUN_UV_VERSION)."; \
+	  exit 1; \
+	fi; \
+	}
+
+.PHONY: update-mlrun-api-deps
+update-api-deps: verify-uv-version ## Update mlrun-api locked requirements file
```
Comment (quaark) [inline Makefile]: The .PHONY name doesn't match the rule name.
Anyway, maybe call it more explicitly `update-mlrun-api-deps-lock` (just to have 'lock' in the name to make it more clear what this does)

**Review by quaark — CHANGES_REQUESTED** [review]
Looks great!
Minor naming issue/suggestion in the makefile

File: Makefile [hunk 3]
```diff
@@ -782,3 +786,23 @@ ifdef MLRUN_DOCKER_CACHE_FROM_TAG
 	done;
     MLRUN_DOCKER_CACHE_FROM_FLAG := $(MLRUN_BASE_IMAGE_DOCKER_CACHE_FROM_FLAG)
 endif
+
+.PHONY: verify-uv-version
+verify-uv-version:
+	@{ \
+	uv_version=$$(uv version | cut -d' ' -f2); \
+	result=$$(pysemver compare $$uv_version $(MLRUN_UV_VERSION)); \
```
Comment (liranbg) [inline Makefile]:
```suggestion
	result=$$(python -m semver compare $$uv_version $(MLRUN_UV_VERSION)); \
```

While this is a suggestion, the reasons for it
1. explicitness
2. indicates exactly which python runs the cli command

File: Makefile (hunk 2 above)
Comment (jond01) [reply Makefile]: Renamed to `upgrade-mlrun-api-deps-lock` and fixed the .PHONY name in 4e98e41

File: Makefile (hunk 3 above)
Comment (jond01) [reply Makefile]: Done in b5db017

**Review by liranbg — APPROVED** [review]

**Review by assaf758 — APPROVED** [review]

**Review by quaark — APPROVED** [review]

**Pull request merged**